internal/controllers/phase_ensure_addon_instance.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+// Ensures the presense or absense of an OperatorGroup depending on the Addon install type.
+func (r *AddonReconciler) ensureAddonInstance(
+	ctx context.Context, log logr.Logger, addon *addonsv1alpha1.Addon) (err error) {
+	// not capturing "stop" because it won't ever be reached due to the guard rails of CRD Enum-Validation Markers
+	targetNamespace, _, _, err := r.parseAddonInstallConfig(ctx, log, addon)
+	if err != nil {
+		return err
+	}
```

Comment:
It seems like we're still capturing the `stop` in all other places - https://github.com/openshift/addon-operator/blob/main/internal/controllers/phase_ensure_catalogsource.go#L39

The invalid configuration will be prevented by CRD validation and the validation webhook.

Reply:
> The invalid configuration will be prevented by CRD validation and the validation webhook.
That's what I am trying to convey :) 
`ensureAddonInstance` is getting triggered only after a successful creation of `Addon` object. But the CRD enum validation marker ensures that at the time of `kubectl apply`-ing the Addon's YAML manifest, if the enum validation fails (the case when stop==true and err==nil), the creation itself of the `Addon` object will be blocked meaning that even `ensureAddonInstance` wouldn't be called. Hence, capturing `stop` in this case sounds like a redundant  case to handle.

Reply:
That's right :) I was just trying to address why we're still capturing `stop` in other places when we have the CRD validation and webhooks stopping it.

Well, as for webhooks, we can still run the controller without it. I'm not sure when the CRD validation can fail, but it probably doesn't harm to add the extra check by capturing `stop` to prevent the controller from panicking in case the schema validation fails. :)